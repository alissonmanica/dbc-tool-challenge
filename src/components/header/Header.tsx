import { useDispatch, useSelector } from 'react-redux'
import { toolActions } from '../../store/tools/duck'
import toolSelector from '../../store/tools/selector'
import Button from '../Button'
import Search from '../Search'
import {
  DivHeaderTools,
  DivSearch,
  HeaderTools,
  CheckBoxSearch,
  SubTitleHeader,
  TitleHeader,
  LabelCheckSearch
} from './Header.styles'

function Header() {
  const dispatch = useDispatch()
  const isOnlyTag = useSelector(toolSelector.isOnlyTag)

  return (
    <HeaderTools>
      <TitleHeader>VUTTR</TitleHeader>
      <SubTitleHeader>Very Useful Tools to Remember</SubTitleHeader>
      <DivHeaderTools>
        <DivSearch>
          <Search />
          <LabelCheckSearch>
            <CheckBoxSearch
              onChange={() => dispatch(toolActions.toggleIsOnlyTag())}
              type="checkbox"
              id="tags-only"
              name="tags-only"
              checked={isOnlyTag}
            />
            search in tags only
          </LabelCheckSearch>
        </DivSearch>
        <Button />
      </DivHeaderTools>
    </HeaderTools>
  )
}
export default Header
