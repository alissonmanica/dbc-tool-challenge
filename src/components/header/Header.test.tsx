import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

import Header from './Header'

describe('snapshot testing', () => {
  const initialState = {
    tools: [],
    toolsSearch: [],
    isOnlyTag: false,
    openModalAdd: false
  }
  const mockStore = configureStore()
  let store = mockStore(initialState)

  it('should test snapshot Header component', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})
