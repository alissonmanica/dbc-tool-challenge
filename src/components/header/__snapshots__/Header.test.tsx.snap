// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should test snapshot Header component 1`] = `
<header
  className="sc-bczRLJ bBPpYY"
>
  <h1
    className="sc-gsnTZi XGVfn"
  >
    VUTTR
  </h1>
  <h2
    className="sc-dkzDqf eKMNHQ"
  >
    Very Useful Tools to Remember
  </h2>
  <div
    className="sc-hKMtZM gifOhm"
  >
    <div
      className="sc-eCYdqJ hGxLib"
    >
      <span
        className="sc-jSMfEi cXCmoX"
      >
        <svg
          className="sc-kDDrLX ZJbUJ"
          fill="none"
          height="20"
          viewBox="0 0 23 20"
          width="23"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.4869 12.5973L22.6591 19.7036"
            stroke="#B1ADB9"
            stroke-miterlimit="10"
            stroke-width="2"
          />
          <path
            d="M8.67798 14.6729C13.2044 14.6729 16.8738 11.4821 16.8738 7.54606C16.8738 3.61001 13.2044 0.41922 8.67798 0.41922C4.15153 0.41922 0.482117 3.61001 0.482117 7.54606C0.482117 11.4821 4.15153 14.6729 8.67798 14.6729Z"
            stroke="#B1ADB9"
            stroke-miterlimit="10"
            stroke-width="2"
          />
        </svg>
        <input
          className="sc-gKXOVf hnGcWT"
          onChange={[Function]}
          placeholder="Search..."
          type="search"
        />
      </span>
      <label
        className="sc-ftvSup kDEeyz"
      >
        <input
          className="sc-iBkjds bQzrOA"
          id="tags-only"
          name="tags-only"
          onChange={[Function]}
          type="checkbox"
        />
        search in tags only
      </label>
    </div>
    <button
      className="sc-papXJ gEqlvR"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="sc-jqUVSM faEMZO"
        fill="none"
        height="24"
        viewBox="0 0 28 24"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.125 13H15.375V19H13.125V13H6.375V11H13.125V5H15.375V11H22.125V13Z"
          fill="white"
        />
      </svg>
      Add
    </button>
  </div>
</header>
`;
